### PR TITLE
Feature/10328 bulk save bugfix

### PR DESCRIPTION
**Pull request by bjss-chris**
Story: [AB#10328](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/10328)
Task: [AB#10335](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/10335)

### CODE REVIEW

File: app/helpers/common/transformApiValidationResponse.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+const transformPropertyName = propertyName => (propertyName.startsWith('[')
+  ? propertyName.substring(4)
+  : propertyName);
+
+export const transformApiValidationResponse = (validationResponse) => {
+  const distinctApiErrors = Object
```
Comment (samMcCarthyBJSS) [inline app/helpers/common/transformApiValidationResponse.js:6]: can the lambdas be jojned to save making an object that's just going to be returned?

File: app/helpers/common/transformApiValidationResponse.test.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
+import { transformApiValidationResponse } from './transformApiValidationResponse';
+
+describe('transformApiValidationResponse', () => {
+  it('should transform a non-bulk validation error response', () => {
+    const field = 'DeliveryDate';
```
Comment (samMcCarthyBJSS) [inline app/helpers/common/transformApiValidationResponse.test.js:5]: could this be a global const?

**Review by samMcCarthyBJSS — COMMENTED** [review]
Left a couple of comments

File: app/helpers/common/transformApiValidationResponse.js (hunk 1 above)
Comment (bjss-chris) [reply app/helpers/common/transformApiValidationResponse.js:6]: Do you mean something like this?

```javascript
export const transformApiValidationResponse = validationResponse => Object.entries(Object
  .entries(validationResponse)
  .reduce((acc, [key, value]) => {
    const field = transformPropertyName(key);

    if (acc[field] === undefined) {
      acc[field] = new Set();
    }

    value.forEach(v => acc[field].add(v));
    return acc;
  }, {}))
  .flatMap(([key, value]) => [...value].map(v => ({ field: key, id: v })));
```
I think the existing version is more readable and I'm not sure we're really gaining much by joining the two expressions. My preference would be to stick with the more readable version.

Comment (samMcCarthyBJSS) [reply app/helpers/common/transformApiValidationResponse.js:6]: yeah did mean that
although was going to suggest then swapping to a foreach or something for the entries bit
but yeah is more readable how you had it

**Review by samMcCarthyBJSS — APPROVED** [review]

**Pull request merged**